torch/autograd/gradcheck.py

```diff
@@ -917,8 +979,10 @@ def _test_backward_mul_by_grad_output(outputs, inputs, check_sparse_nnz) -> bool
                 raise GradcheckError('backward not multiplied by grad_output')
         elif not gi.eq(0).all():
             raise GradcheckError('backward not multiplied by grad_output')
-        if gi.dtype != di.dtype or gi.device != di.device or gi.is_sparse != di.is_sparse:
```

Comment:
So now we are officially ok with tensor and gradient not to have the same sparsity? :tada:

Reply:
Related discussion: https://github.com/pytorch/pytorch/pull/94728#discussion_r1106158960